### PR TITLE
fix: copy .env file error on Windows os.

### PR DESCRIPTION
**Pull request by jeff87218**
## 📑 Description

Replace the command with Go code for greater portability.
Closes [https://github.com/goravel/goravel/issues/461](https://github.com/goravel/goravel/issues/461)  

@coderabbitai summary

### CODE REVIEW

**Comment by hwbrzzl** [comment]
Thanks! Checking...

File: console/commands/new_command.go [hunk 1]
```diff
@@ -185,3 +186,23 @@ func (receiver *NewCommand) removeFiles(path string) error {
 	// Remove the .GitHub directory
 	return file.Remove(filepath.Join(path, ".github"))
 }
+
+func (receiver *NewCommand) copyFile(inputFilePath, outputFilePath string) (err error) {
```
Comment (hwbrzzl) [inline console/commands/new_command.go:190]: Please add a test case for this function.

**Review by hwbrzzl — COMMENTED** [review]
Great PR, thanks Bro! Could you optimize the existing test case like below?

image

File: console/commands/new_command.go (hunk 1 above)
Comment (jeff87218) [reply console/commands/new_command.go:190]: I have updated the unit tests. Please check them out.

File: console/commands/new_command_test.go [hunk 2]
```diff
@@ -47,6 +48,12 @@ func TestNewCommand(t *testing.T) {
 	assert.Contains(t, captureOutput, "App key generated successfully!")
 	assert.True(t, file.Exists("example-app"))
 
+	// Test copyFile
+	src := filepath.Join("example-app", ".env.example")
+	dst := filepath.Join("example-app", ".env")
+	assert.Nil(t, newCommand.copyFile(src, dst))
+	assert.True(t, file.Exists(dst))
```
Comment (hwbrzzl) [inline console/commands/new_command_test.go:55]: This can be moved to a new function: `TestCopyFile`.

File: console/commands/new_command_test.go [hunk 3]
```diff
@@ -47,6 +48,12 @@ func TestNewCommand(t *testing.T) {
 	assert.Contains(t, captureOutput, "App key generated successfully!")
 	assert.True(t, file.Exists("example-app"))
```
Comment (hwbrzzl) [inline console/commands/new_command_test.go:49]:
```suggestion
	assert.True(t, file.Exists("example-app"))
	assert.True(t, file.Exists("example-app/.env"))
```

**Review by hwbrzzl — COMMENTED** [review]
Thanks! Two nitpicks.

**Comment by jeff87218** [comment]
> Thanks! Two nitpicks.

I have updated the unit tests. Please check them out. Thank you for your review.

File: console/commands/new_command_test.go [hunk 4]
```diff
@@ -56,3 +59,20 @@ func TestNewCommand(t *testing.T) {
...
+	src := filepath.Join(tmpDir, ".env.example")
+	dst := filepath.Join(tmpDir, ".env")
+
+	// Create a mock .env.example file for testing
+	err = os.WriteFile(src, []byte("example env"), os.ModePerm)
+	assert.Nil(t, err)
+	assert.True(t, file.Exists(src))
+	assert.Nil(t, newCommand.copyFile(src, dst))
+	assert.True(t, file.Exists(dst))
+	assert.Nil(t, os.Remove(dst))
```
Comment (hwbrzzl) [inline console/commands/new_command_test.go:77]: Please remove `src`, too.

Comment (jeff87218) [reply console/commands/new_command_test.go:78]: sure!

**Review by hwbrzzl — APPROVED** [review]
Great PR, thanks for your contribution! will add you to the list. 👍

**Comment by hwbrzzl** [comment]
Hey @jeff87218  We created a private channel in Discord for contributors. You can click [the link](https://discord.gg/dMs73mSa) and DM me (@Bowen) to join it if you are interested.

Here you can:

1. Send PR link or review others' PRs;
2. Discuss new features before creating an issue;
3. Vote for something;
4. etc.

**Pull request merged**